packages/gatsby-core-utils/src/service-lock.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Service lock: handles service discovery for Gatsby develop processes
+ * The problem:  the develop process starts a proxy server, the actual develop process and a websocket server for communication. The two latter ones have random ports that need to be discovered. We also cannot run multiple of the same site at the same time.
+ * The solution: lockfiles! We create a lockfolder in `.config/gatsby/sites/${sitePathHash} and then write a file to that lockfolder for every service with its port.
+ */
+import path from "path"
+import os from "os"
+import crypto from "crypto"
+import lockfile from "proper-lockfile"
+import fse from "fs-extra"
+
+const globalConfigPath =
```

Comment:
Let's use `Configstore` like we do in telemetry. 

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-telemetry/src/event-storage.js#L5

Reply:
This is how the module that is used by configstore to determine the globalConfigPath ([`xdg-basedir`](https://www.npmjs.com/package/xdg-basedir)) works, that's where we "copied" that code from:

https://github.com/sindresorhus/xdg-basedir/blob/a8b4130c61413006f1716d84b92a5ffd6e9336e8/index.js#L11-L12

The reason we don't use that module directly is because it has a fallback to `undefined` in case `os.homedir()` doesn't exist. However, looking at [the Node.js docs](https://nodejs.org/api/os.html#os_os_homedir), `os.homedir()` is always a string, so we inlined the code and fixed that to make TypeScript happy.